src/proteingym/base/dataset.py

```diff
@@ -588,13 +588,20 @@ def to_df(
             ~pl.all_horizontal([pl.col(t).is_null() for t in target_names])
         )
 
-        # Group by sequence and variables, and aggregate the target by mean
+        # Group by sequence and variables, and aggregate the target by mean,
```

Comment:
as discussed - another option is to have `aggfunc` and argument, which maybe a function that does you list below but also enables one to do something else

Also good log a warning if particularly the `.first` strategy is ever used since it lossy and probably most of the time a workaround rather than a solution.